### PR TITLE
chore(views): Add headline for adding unique identifier

### PR DESCRIPTION
**Pull request by janpio**
https://docs-git-janpio-views-unique-identifier-prisma.vercel.app/concepts/components/prisma-schema/views#adding-a-unique-identifier-to-an-introspected-view
image

### CODE REVIEW

**Review by Druue — APPROVED** [review]

File: content/200-concepts/100-components/01-prisma-schema/08-views.mdx [hunk 1]
```diff
@@ -253,7 +253,17 @@ view UserInfo {
...
-In this view's case, the `id` column refers to a uniquely identifiable field in the underlying `User` table so that field will also be used as the uniquely identifiable field in the `view` block.
+<Admonition type="warning">
+
+Please note for now `db pull` will only introspect views in your schema when using PostgreSQL. Support for this workflow will be extended to other database providers soon.
+
+</Admonition>
+
+#### Adding a unique identifier to an introspected view
+
+To be able to use the introspected view in Prisma Client, you will need to select and define one or multiple of the fields as the unique identifier.
```
Comment (Druue) [inline content/200-concepts/100-components/01-prisma-schema/08-views.mdx:264]: I would maybe update this to say "as a unique identifier" due to the multiple

Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/08-views.mdx:264]: Hmm grammar question. The "multiple of the fields" refers to `@@unique[field1, field2]`. Is it then more correct to say "a unique identifier"? I honestly don't know.

Comment (Druue) [reply content/200-concepts/100-components/01-prisma-schema/08-views.mdx:264]: ah, oh no, hmm, as a tech q, does `@@unique([...])` refer to: individual fields not needing to be unique in and of themselves but the combination _does_ ? or is it just a set of unique identifiers

hopefully this informs something re: grammar

File: content/200-concepts/100-components/01-prisma-schema/08-views.mdx [hunk 2]
```diff
@@ -253,7 +253,17 @@ view UserInfo {
 
 The `view` block is generated initially with a `@@ignore` attribute because [there is no unique identifier defined](#unique-identifier) (which is currently a [limitation](#unique-identifier) of the views preview feature).
 
-In this view's case, the `id` column refers to a uniquely identifiable field in the underlying `User` table so that field will also be used as the uniquely identifiable field in the `view` block.
+<Admonition type="warning">
+
+Please note for now `db pull` will only introspect views in your schema when using PostgreSQL. Support for this workflow will be extended to other database providers soon.
```
Comment (pimeys) [inline content/200-concepts/100-components/01-prisma-schema/08-views.mdx]: Should we include CDB, MySQL and SQL Server here too?

File: content/200-concepts/100-components/01-prisma-schema/08-views.mdx (hunk 1 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/08-views.mdx:264]: The combination of the multiple fields need to be unique, yes. We sometimes call the "compound uniques".

If you have mutliple fields that need to be unique by themselves, then you use two `field1 Type @unique` and `field2 Type @unique` instead, so via attributes on the field directly.

File: content/200-concepts/100-components/01-prisma-schema/08-views.mdx (hunk 2 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/08-views.mdx]: Yes, but @sabinadams will do that via a separate PR to make sure all mentions are updated. I am just separating out the section in this PR.

File: content/200-concepts/100-components/01-prisma-schema/08-views.mdx (hunk 1 above)
Comment (Druue) [reply content/200-concepts/100-components/01-prisma-schema/08-views.mdx:264]: Ah, okay, in the case of `@@unique` then "the unique identifier" makes most sense to me

**Review by sabinadams — APPROVED** [review]

**Pull request merged**